Source/Entities/Validation/ValidationRules.cst

```diff
@@ -87,6 +95,27 @@ namespace <%=NameSpace%>.Validation
             return _rulesList;
          }
       }
+	  
+	  /// <summary>
+	  /// List of common rules associated with an entity.
+	  /// </summary>
+	  private static ConcurrentDictionary<string, ConcurrentDictionary<string, List<ValidationRuleInfo>>> CommonRulesList
+      {
+         get

```

Comment:
Static initialisation of the private member might be simpler than this double-lock checking.